File: src/main/java/com/illuzionzstudios/mist/scheduler/rate/Async.java
package com.illuzionzstudios.mist.scheduler.rate;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD, ElementType.FIELD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface Async {

    /**
     * @return The rate of refresh
     */
    Rate rate() default Rate.INSTANT;
}

File: src/main/kotlin/com/illuzionzstudios/mist/scheduler/rate/Async.kt
package com.illuzionzstudios.mist.scheduler.rate

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS
)
@Retention(
    RetentionPolicy.RUNTIME
)
annotation class Async(
    /**
     * @return The rate of refresh
     */
    val rate: Rate = Rate.INSTANT
)